ams/server/src/main/java/com/netease/arctic/server/table/TableSelfOptimizingMetrics.java

```diff
@@ -0,0 +1,350 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netease.arctic.server.table;
+
+import static com.netease.arctic.api.metrics.MetricDefine.defineCounter;
+import static com.netease.arctic.api.metrics.MetricDefine.defineGauge;
+
+import com.netease.arctic.api.metrics.Counter;
+import com.netease.arctic.api.metrics.Gauge;
+import com.netease.arctic.api.metrics.Metric;
+import com.netease.arctic.api.metrics.MetricDefine;
+import com.netease.arctic.api.metrics.MetricKey;
+import com.netease.arctic.server.metrics.MetricRegistry;
+import com.netease.arctic.server.optimizing.OptimizingStatus;
+import com.netease.arctic.server.optimizing.OptimizingType;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+
+import java.util.List;
+
+/** Table self optimizing metrics */
+public class TableSelfOptimizingMetrics {
+  /** Table is no need optimizing. */
+  public static final String STATE_IDLE = "idle";
```

Comment:
STATE seems not quite proper here.

In process api these are named stage(we could find similar concepts in Spark)

Reply:
I change the field to `STATUS_xxx` and it represents OptimizingStatus. (But in the metric design, we use `executing` instead of `minor`, `major`, `full`)

When we design the table optimzing metrics, we all agree that `minor`, `major`, `full` is a optimizing process type, not a status.